### PR TITLE
Add an options for viewing the plain text if available

### LINKED ISSUE #906: Add option for default display format
Allow the user to display `text/plain` parts of emails by default.

### PR DESCRIPTION
**Pull request by philipwhiuk**
Probably some bike-shedding to be done regarding the name of the preference in code (displayPreference is what I came up with at the time - there must be better). But at least it gets the ball rolling.

Defaults to HTML preferred because I don't want the default behaviour for new users be an email app that looks like it's only capable of plain text mail.

Will fix #906

### CODE REVIEW

**Comment by cketti** [comment]
I think the body loading part should fetch the desired part and prepare it for being displayed in the WebView. Passing a lot of data and a flag through a bunch of layers feels very wrong.

File: k9mail/src/main/java/com/fsck/k9/view/MessageWebView.java [hunk 1]
```diff
@@ -112,8 +112,9 @@ private void disableDisplayZoomControls() {
      *
      * @param text
      *      The message body to display.  Assumed to be MIME type text/html.
+     * @param isHtml
```
Comment (Valodim) [inline k9mail/src/main/java/com/fsck/k9/view/MessageWebView.java]: classic example of why we don't document ;)

File: k9mail/src/main/java/com/fsck/k9/Account.java [hunk 2]
```diff
@@ -198,6 +198,7 @@ public boolean isDefaultAscending() {
     private SortType mSortType;
     private Map<SortType, Boolean> mSortAscending = new HashMap<SortType, Boolean>();
     private ShowPictures mShowPictures;
+    private DisplayPreference mDisplayPreference;
```
Comment (Valodim) [inline k9mail/src/main/java/com/fsck/k9/Account.java]: m prefix no longer in code style

**Comment by Valodim** [comment]
I agree with cketti. Also, sorry for the conflicts from the quotedmessagepresenter :(

**Comment by philipwhiuk** [comment]
I'm going to close this and come back to it when I have time. There's an open issue for it so give this needs some considerable rework I see no reason for it to remain open.

File: k9mail/src/main/java/com/fsck/k9/Account.java (hunk 2 above)
Comment (francwalter) [inline k9mail/src/main/java/com/fsck/k9/Account.java]: > @Valodim  m prefix no longer in code style

Where can I read about this?

Comment (cketti) [inline k9mail/src/main/java/com/fsck/k9/Account.java]: https://github.com/k9mail/k-9/wiki/CodeStyle

File: k9mail/src/main/java/com/fsck/k9/mailstore/MessageViewInfo.java [hunk 3]
```diff
@@ -19,15 +19,15 @@ public MessageViewInfo(List<MessageViewContainer> containers, Message message) {
...
     public static class MessageViewContainer {
-        public final String text;
+        public final ViewableContainer viewable;
         public final Part rootPart;
         public final List<AttachmentViewInfo> attachments;
         public final OpenPgpResultAnnotation cryptoAnnotation;
 
 
-        MessageViewContainer(String text, Part rootPart, List<AttachmentViewInfo> attachments,
+        MessageViewContainer(ViewableContainer viewable, Part rootPart, List<AttachmentViewInfo> attachments,
```
Comment (dhebbeker) [inline k9mail/src/main/java/com/fsck/k9/mailstore/MessageViewInfo.java]: Replying to @cketti's [comment](https://github.com/k9mail/k-9/pull/1212#issuecomment-215605596):
> Passing a lot of data and a flag through a bunch of layers feels very wrong.

By "a lot of data", do you mean the overhead of the full `viewable` in contrast to `text`?

**Pull request closed without merging**